leader/logger.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+// Borrowed from https://github.com/hashicorp/consul-api-gateway/blob/main/internal/k8s/logger.go
```

Comment:
nit: Could we replace `main` with a commit hash so the link never(-ish!) bitrots?